fdbserver/storageserver.actor.cpp

```diff
@@ -2680,13 +2689,15 @@ ACTOR Future<std::pair<ChangeFeedStreamReply, bool>> getChangeFeedMutations(Stor
 		}
 
 		wait(data->changeFeedDiskReadsLock.take(TaskPriority::DefaultYield));
+		state PriorityMultiLock::Lock ssReadLock = wait(data->getReadLock(req.options));
```

Comment:
My understanding of flow locks could certainly be wrong, but doesn't the holdingDiskReads lock releaser need to be constructed immediately after the take returns, before the priority lock is taken?
Otherwise if the actor is cancelled when waiting for the priority multi lock, we would leak a change feed disk reads slot that wouldn't get released, because there was no releaser?
It's also possible that the existence of the priority multi lock here can supercede the original change feed disk reads lock, since they will both limit the parallelism of this section

Reply:
Correct on the releaser, I'll fix that if we keep the lock.  I didn't hit any failures for that but probably because when this actor is cancelled the lock is also being destroyed because a shutdown is happening.

I was also thinking the change feed disk read lock should not be necessary but forgot to follow up on it, got distracted with debugging.  I don't see a reason to have it, with the ssLock these requests will be triaged and get a share of concurrency based on their priority.  BlobGranule change feed reads will use the Normal read type and fetches will use the Fetch read type, and we can further isolate change feeds if we want using the read type and ssLock configuration.

Reply:
Change feed actors can get cancelled at other points though. Maybe buggifying the lock count down would have helped find it?
The primary purpose for this lock originally was to have a limit here to avoid building up uncancellable reads in sqlite iirc. The priority multi lock would likely serve the same purpose

Reply:
One thing the lock provides is a count of concurrent change feed disk requests, but I'm not sure how useful of a metric this.  Looking at the number of change feed requests per 5s interval seems sufficient?

Reply:
In this PR there is logging of waiters and active requests per ReadType but it's not perfect as we can only get these numbers for each Priority within the PML, and the configuration can and does map multiple ReadTypes to a single Priority.

Reply:
Yeah, i've just been looking at the change feed disk reads counter rather than this lock